File: fragment/Mess_from_activitytofragment2/app/src/main/java/com/example/mess_from_activitytofragment2/MainActivity.java
package com.example.mess_from_activitytofragment2;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private EditText medit;
    Button button;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        medit = findViewById(R.id.etusername);
        button = findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                firstFragment first = new firstFragment();
                FragmentTransaction tra = getSupportFragmentManager().beginTransaction();
                tra.replace(R.id.frag_container,first);
                tra.commit();

                String mess = medit.getText().toString();

                first.setData(mess);

            }
        });
    }

}

File: binding/androidViewBindingFragment/app/src/main/java/com/example/androidviewbindingfragment/MainActivity.kt
package com.example.androidviewbindingfragment

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import com.example.androidviewbindingfragment.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        //setContentView(R.layout.activity_main)

        val binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        //note all id here in camel case removing space in between them
        with(binding){
            tvOutput1.text = "Hello World!"
            //made nullable because in landscape it is not present
            tvOutput2?.text = "Welcome to View Binding"
            includedLayout?.tvIncludeOutput1?.text = "View Binding in Included layout..."

            btnClickMe.setOnClickListener {
                val fragment = MainFragment()

                supportFragmentManager
                    .beginTransaction()
                    .add(R.id.fragment_container,fragment)
                    .commit()
            }
        }
    }
}